# 8a

- Recency: 0/1
- Duration: 0/1
- Genre: 0/1/2/3/4

= 3 different attributes


Different combinations $= P = 2\cdot 2 \cdot 5 = 20$

# 8b

Genres not mutually exclusive:


You can choose a 0 or 1 across all genres $\rightarrow$ 7 attributes:

$$P_1 = 2^7 = 128$$

In [1]:
print(2**7)

128


128 is a lot more than 20 and does represent a more realistic picture of the very varying nature of video content that always has more or less of each genre, and different durations and recency. 20 different product versions is a generalization that is possible to work with for the conjoint analysis

# 8c

In [2]:
import pandas as pd
from pandas import DataFrame

In [4]:
def get_user_data() -> DataFrame:
    data = pd.read_csv('../filter_user.csv', sep=';')
    return data

In [9]:
user_data = get_user_data()
user_data['combination'] = user_data.apply(
    lambda row: f"{row['recency_new']}_{row['duration_long']}_" + 
                "_".join([col for col in user_data.columns if col.startswith('genre_') and row[col] == 1]),
    axis=1
)
combination_counts = user_data['combination'].value_counts()

print(combination_counts)


0_1_genre_family_comedy       5
1_0_genre_family_comedy       2
1_1_genre_family_comedy       2
0_1_genre_action_adventure    2
1_1_genre_drama               2
1_1_genre_documentary         1
0_1_genre_drama               1
0_0_genre_documentary         1
1_1_genre_action_adventure    1
0_0_genre_family_comedy       1
1_0_genre_thriller_crime      1
1_1_genre_thriller_crime      1
Name: combination, dtype: int64


In the rankings of the combinations of attribute levels there is a clear winner with 5 instances, and that is non-recent long family comedies. This user likes family comedies, as the two next most popular with 2 instances each are also family comedies, but these are recent short, and recent long

In [12]:
family_comedy = 10/4
action_adventure = 3/2
drama = 3/2
documentary = 2/2
thriller_crime = 2/2

recency_new = 10/7
recency_old = 10/5

duration_long = 15/8
duration_short = 5/4

all_avg_scores = [family_comedy, action_adventure, drama, documentary, thriller_crime, recency_new, recency_old, duration_long, duration_short]

In [14]:
maximum_score = max(all_avg_scores)
minimum_score = min(all_avg_scores)

PWUs = [round((score - minimum_score) / (maximum_score - minimum_score), 2) for score in all_avg_scores]
print(PWUs)

[1.0, 0.33, 0.33, 0.0, 0.0, 0.29, 0.67, 0.58, 0.17]


In [17]:
PWUs_genre= PWUs[:5]
raw_importance_genre = max(PWUs_genre) - min(PWUs_genre)
PWUs_recency = PWUs[5:7]
raw_importance_recency = max(PWUs_recency) - min(PWUs_recency)
PWUs_duration = PWUs[7:]
raw_importance_duration = max(PWUs_duration) - min(PWUs_duration)

print("raw_importance_genre: ", raw_importance_genre)
print("raw_importance_recency: ", raw_importance_recency)
print("raw_importance_duration: ", raw_importance_duration)

raw_importance_genre:  1.0
raw_importance_recency:  0.38000000000000006
raw_importance_duration:  0.4099999999999999


In [18]:
sum_raw_importance = raw_importance_genre + raw_importance_recency + raw_importance_duration

relative_importance_genre = raw_importance_genre / sum_raw_importance
relative_importance_recency = raw_importance_recency / sum_raw_importance
relative_importance_duration = raw_importance_duration / sum_raw_importance
print("relative_importance_genre: ", relative_importance_genre)
print("relative_importance_recency: ", relative_importance_recency)
print("relative_importance_duration: ", relative_importance_duration)

relative_importance_genre:  0.5586592178770949
relative_importance_recency:  0.21229050279329612
relative_importance_duration:  0.2290502793296089


For this user the genre is most important with a relative importance of 0.56. Recency and duration has the almost same relative importance with duration being a marginally more important with 0.23 vs. 0.21, while genre is more than twice as important as the two other attributes. This theory is supported by the dominance of family comedies for this specific user, while the other attributes have less of a pattern in the data compared to the love for family comedies